openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -282,23 +294,140 @@ def _update_plugin_configuration(
         """
         Create/update legacy provider settings
         """
+        plugin_configuration = validated_data.pop('plugin_configuration', {})
         updated_provider_type = validated_data.get('provider_type') or instance.provider_type
         will_support_legacy = bool(
             updated_provider_type == 'legacy'
         )
         if will_support_legacy:
-            course_key = instance.context_key
-            course = get_course_by_id(course_key)
             legacy_settings = LegacySettingsSerializer(
-                course,
+                self._get_course(),
                 context={
                     'user_id': self.context['user_id'],
                 },
-                data=validated_data.get('plugin_configuration', {}),
+                data=plugin_configuration,
             )
             if legacy_settings.is_valid(raise_exception=True):
                 legacy_settings.save()
-            instance.plugin_configuration = {}
+            instance.plugin_configuration = {
+                "group_at_subsection": plugin_configuration.get("group_at_subsection", False)
+            }
         else:
-            instance.plugin_configuration = validated_data.get('plugin_configuration') or {}
+            instance.plugin_configuration = plugin_configuration
+        return instance
+
+    def _update_course_configuration(
+        self,
+        instance: DiscussionsConfiguration,
+        validated_data: dict,
+    ) -> DiscussionsConfiguration:
+        """
+        Update configuration settings that are stored in the course.
+        """
+        save = False
+        updated_provider_type = validated_data.get('provider_type') or instance.provider_type
+        for key in self.Meta.course_fields:
+            value = validated_data.get(key)
+            # Delay loading course till we know something has actually been updated
+            if value is not None and value != getattr(instance, key):
+                self._get_course().discussions_settings[key] = value
+                save = True
+        new_plugin_config = validated_data.get('plugin_configuration', None)
+        if new_plugin_config and new_plugin_config != instance.plugin_configuration:
+            save = True
+            # Any fields here that aren't already stored in the course structure
+            # or in other models should be stored here.
+            self._get_course().discussions_settings[updated_provider_type] = {
+                key: value
+                for key, value in new_plugin_config.items()
+                if (
+                    key not in LegacySettingsSerializer.Meta.fields and
+                    key not in LegacySettingsSerializer.Meta.fields_cohorts
+                )
+            }
```

Comment:
Can you explain the reason why all discussion configs are stored in the course structure?